code/game/machinery/dna_infuser/dna_infuser.dm

```diff
@@ -163,6 +192,10 @@
 	if(user.stat != CONSCIOUS || HAS_TRAIT(user, TRAIT_UI_BLOCKED) || !Adjacent(user) || !user.Adjacent(target) || !ISADVANCEDTOOLUSER(user))
 		return
 
+	if(iscarbon(target))
+		close_machine(target)
```

Comment:
why are we closing the machine before we check if they are valid to infuse? They arent gonna be forcemoved into it already

Reply:
There won't be any carbons you can infuse into someone, but all subjects are carbo- actually, HUMANS are subjects, so let me change this